packages/@lwc/module-resolver/src/index.ts

```diff
@@ -6,154 +6,113 @@
-            const rootDir = path.dirname(file);
-            const rootParts = rootDir.split('/'); // the glob library normalizes paths to forward slashes only - https://github.com/isaacs/node-glob#windows
-            const entry = path.join(absPath, file);
-
-            const dirModuleName = rootParts.pop();
-            const dirModuleNamespace = rootParts.pop();
-            if (dirModuleNamespace && dirModuleName === fileName) {
-                const registryNode = createRegistryEntry(
-                    entry,
-                    `${dirModuleNamespace}/${fileName}`,
-                    fileName,
-                    dirModuleNamespace.toLowerCase()
-                );
-                mappings[registryNode.moduleSpecifier] = registryNode;
-            }
-
-            return mappings;
-        }, {});
+    return resolvedModules;
 }
 
-function hasModuleBeenVisited(module, visited) {
-    return visited.has(module);
-}
+export function resolveModulesFromNpm(packageName: string): RegistryEntry[] {
+    let resolvedModules: RegistryEntry[] = [];
+    try {
+        const pkgJsonPath = require.resolve(`${packageName}/package.json`);
+        const packageDir = path.dirname(pkgJsonPath);
+        const lwcConfigFile = path.join(packageDir, LWC_CONFIG_FILE);
 
-function expandModuleDirectories({
-    moduleDirectories,
-    rootDir,
-    modulePaths,
-}: Partial<ModuleResolverConfig> = {}) {
-    if (modulePaths) {
-        return modulePaths;
-    }
-    if (!moduleDirectories && !rootDir) {
-        // paths() is spec'd to return null only for built-in node
-        // modules like 'http'. To be safe, return empty array in
-        // instead of null in this case.
-        return require.resolve.paths('.') || [];
+        if (fs.existsSync(lwcConfigFile)) {
+            resolvedModules = resolveModules({ rootDir: lwcConfigFile });
+        } else {
+            const pkgJson = JSON.parse(fs.readFileSync(pkgJsonPath, 'utf8'));
+            if (pkgJson.lwc) {
+                resolvedModules = resolveModulesFromList(pkgJson.lwc.modules, { root: packageDir });
+            }
+        }
+    } catch (e) {
+        /*noop*/
     }
 
-    return nodeModulePaths(rootDir || __dirname, moduleDirectories);
+    return resolvedModules;
 }
 
-function resolveModules(modules, opts) {
-    if (Array.isArray(modules)) {
-        modules.forEach(modulePath => resolveModules(modulePath, opts));
-    } else {
-        const { mappings, visited, moduleRoot } = opts;
-        if (typeof modules === 'string') {
-            const packageEntries = resolveModulesInDir(path.join(moduleRoot, modules));
-            Object.keys(packageEntries).forEach(moduleName => {
-                if (!hasModuleBeenVisited(moduleName, visited)) {
-                    mappings[moduleName] = packageEntries[moduleName];
-                    visited.add(moduleName);
-                }
-            });
+export function resolveModulesFromList(
```

Comment:
I don't think we should expose this API out of the package, the `resolveModules` API should be sufficient.